core/src/main/java/io/snappydata/impl/SnappyHiveCatalog.java

```diff
@@ -102,17 +114,23 @@ public static synchronized String initCommonHiveMetaStoreProperties(
     metadataConf.set("datanucleus.mapping.Schema", Misc.SNAPPY_HIVE_METASTORE);
     // Tomcat pool has been shown to work best but does not work in split mode
     // because upstream spark does not ship with it (and the one in snappydata-core
-    //   cannot be loaded by datanucleus which should apparently be in its CLASSPATH)
-    metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_POOLING_TYPE, "BONECP");
-    String warehouse = metadataConf.get(HiveConf.ConfVars.METASTOREWAREHOUSE.varname);
-    if (warehouse == null || warehouse.isEmpty() ||
-        warehouse.equals(HiveConf.ConfVars.METASTOREWAREHOUSE.getDefaultExpr())) {
-      // append warehouse to current directory
-      warehouse = new java.io.File("./warehouse").getAbsolutePath();
-      metadataConf.setVar(HiveConf.ConfVars.METASTOREWAREHOUSE, warehouse);
-    }
+    //   cannot be loaded by datanucleus which should be in system CLASSPATH).
+    // Using inbuilt DBCP pool which allows setting the max time to wait
+    // for a pool connection else BoneCP hangs if network servers are down, for example,
+    // and the thrift JDBC connection fails since its default timeout is infinite.
+    // The DBCP 1.x versions are thoroughly outdated and should not be used but
+    // the expectation is that the one bundled in datanucleus will be in better shape.
+    metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_POOLING_TYPE,
+        "dbcp-builtin");
+    // set the scratch dir inside current working directory (unused but created)
+    setDefaultPath(metadataConf, HiveConf.ConfVars.SCRATCHDIR, "./hive");
```

Comment:
Should we create a '.hive' folder inside the cwd and use that and also ./warehouse inside it. Users need to be notified of this in the docs? Any implication on backup scripts?

Reply:
These are unused directories and remain empty so no docs or backup implication. The change is to avoid issues where user may not have permissions in default locations (which is required during hive initialization).